apps/gdalalg_raster_proximity.cpp

```diff
@@ -0,0 +1,199 @@
+    AddArg("distance-units", 0, _("Distance units"), &m_distanceUnits)
+        .SetChoices("pixel", "geo")
+        .SetDefault(m_distanceUnits);
+    AddArg("max-distance", 0,
+           _("Maximum distance. The nodata value will be used for pixels "
+             "beyond this distance"),
+           &m_maxDistance)
+        .SetDefault(m_maxDistance);
+    AddArg("fixed-buffer", 0,
+           _("Fixed buffer value (instead of the actual distance)"),
+           &m_fixedBufferValue)
+        .SetMinValueIncluded(0)
+        .SetDefault(m_fixedBufferValue);
+    AddArg("nodata", 0,
+           _("Specify a nodata value to use for pixels that are beyond the "
+             "maximum distance"),
+           &m_noDataValue);
+}
+
+/************************************************************************/
+/*                 GDALRasterProximityAlgorithm::RunImpl()              */
+/************************************************************************/
+
+bool GDALRasterProximityAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
+                                           void *pProgressData)
+{
+
+    auto srcDS = m_inputDataset.GetDatasetRef();
+    CPLAssert(srcDS);
+
+    const auto srcBand = srcDS->GetRasterBand(m_inputBand);
+    // Check done by GDALAlgorithm::AddBandArg()
+    CPLAssert(srcBand);
+
+    auto dstDs = m_outputDataset.GetDatasetRef();
+
+    if (!dstDs && !m_outputDataset.GetName().empty())
+    {
+        const std::string outputFormat{
+            m_outputFormat.empty()
+                ? GetOutputDriverForRaster(m_outputDataset.GetName().c_str())
+                : m_outputFormat};
+
+        const auto drv =
+            GDALDriver::FromHandle(GDALGetDriverByName(outputFormat.c_str()));
+        if (!drv)
+        {
+            ReportError(CE_Failure, CPLE_AppDefined,
+                        "Output driver '%s' not found.", outputFormat.c_str());
+            return false;
+        }
+        // Create the output dataset with the specified type
+        CPLStringList creationOptions{m_creationOptions};
+
+        GDALDataType outputType = GDT_Float32;
+        if (!m_outputDataType.empty())
+        {
+            outputType = GDALGetDataTypeByName(m_outputDataType.c_str());
+        }
+
+        dstDs = GDALDataset::FromHandle(drv->Create(
+            m_outputDataset.GetName().c_str(), srcBand->GetXSize(),
+            srcBand->GetYSize(), 1, outputType, creationOptions.List()));
+
+        if (!dstDs)
+        {
+            ReportError(CE_Failure, CPLE_AppDefined,
+                        "Failed to create output dataset '%s'.",
+                        m_outputDataset.GetName().c_str());
+            return false;
+        }
+
+        double adfGeoTransform[6];
+        if (srcDS->GetGeoTransform(adfGeoTransform) == CE_None)
+        {
+            dstDs->SetGeoTransform(adfGeoTransform);
+        }
+        dstDs->SetProjection(srcDS->GetProjectionRef());
```

Comment:
avoids going through string serialization:

```suggestion
        dstDs->SetSpatialRef(srcDS->GetSpatialRef());
```

Reply:
done